### DISCUSSION TITLE
Customer Server, useFileSystemPublicRoutes, and the pages folder

### DISCUSSION THREAD

**Discussion by austinthetaco** [vercel/next.js#11218] [Help]
### A Few Things

- I can't seem to find the documentation on the render() method anywhere. It's used in the custom server info but it doesn't actually say what the arguments are or how to use it. 
- The docs say to set useFileSystemPublicRoutes to false if you don't want to use `/pages`, but then how do you point the render method to the right folder/file? How do you actually route to places? 

I couldn't find documentation on either. My whole goal is to not use the `/pages` folder and to be able to write custom routes.

**Comment by austinthetaco** [comment]
bump

**Comment by logusgraphics** [comment]
Same issue. Very unclear on how to disable file system. Even after setting useFileSystemPublicRoutes to false, route based file system files are attempted to be loaded.

**Comment by msreekm** [comment]
agree, need more clarification/documentation on this.struggling with the custom server and routing.
